crates/miden-objects/src/batch/proven_batch.rs

```diff
@@ -115,3 +115,43 @@ impl ProvenBatch {
         &self.output_notes_smt
     }
 }
+
+// SERIALIZATION
+// ================================================================================================
+
+impl Serializable for ProvenBatch {
+    fn write_into<W: ByteWriter>(&self, target: &mut W) {
+        self.id.write_into(target);
+        self.reference_block_commitment.write_into(target);
+        self.reference_block_num.write_into(target);
+        self.account_updates.write_into(target);
+        self.input_notes.write_into(target);
+        self.output_notes_smt.write_into(target);
+        self.output_notes.write_into(target);
+        self.batch_expiration_block_num.write_into(target);
+    }
+}
+
+impl Deserializable for ProvenBatch {
+    fn read_from<R: ByteReader>(source: &mut R) -> Result<Self, DeserializationError> {
+        let id = BatchId::read_from(source)?;
+        let reference_block_commitment = Digest::read_from(source)?;
+        let reference_block_num = BlockNumber::read_from(source)?;
+        let account_updates = BTreeMap::read_from(source)?;
+        let input_notes = InputNotes::<InputNoteCommitment>::read_from(source)?;
+        let output_notes_smt = BatchNoteTree::read_from(source)?;
+        let output_notes = Vec::<OutputNote>::read_from(source)?;
+        let batch_expiration_block_num = BlockNumber::read_from(source)?;
```

Comment:
Similar to the other comment: this assumes that the serialization is trusted. We should probably do some validation here (or maybe move validation to `Self::new()` and have `new_unchecked()` method as well?

Reply:
I guess there isn't a lot we can validate here. Eventually, `ProvenBatch` will be accompanied with a ZKP that validates it - but for now we don't have it.

A couple of things that can be validated:
- We can make sure that `output_notes_smt` and `output_notes` are consistent - i.e., every note in `output_notes` is a leaf in `output_notes_smt` (and there are no extra leaves).
- `batch_expiration_block_num` > `reference_block_num`.

I'd probably do these checks here and return a deserialization error rather than change now `Self::new()` works.

Reply:
But also, we can do these in a different PR. Let's just create an issue for that.

Reply:
I created https://github.com/0xPolygonMiden/miden-base/issues/1150 .

Reply:
It is meant to address this and the unsafe de/serialization of `ProposedBatch` mentioned in above's comment (https://github.com/0xPolygonMiden/miden-base/pull/1140#discussion_r1953094219)